src/components/Clock.js

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable react/button-has-type */
+/* eslint-disable consistent-return */
+
+import React from 'react';
+import './Clock.scss';
+
+export class Clock extends React.Component {
+  state = {
+    date: new Date(),
+    name: '$',
+    visibility: false,
+    previousName: '1',
```

Comment:
store only current name in the state. You can print previous name in console while setting it inside setState